test/test_evaluator.cpp

```diff
@@ -105,6 +113,21 @@ TEST_F(ExprEvalTest, Bindings) {
   checkIntValue(evaluator, d, -2);
 }
 
+// Evaluate known values with const expression evaluator reference
+TEST_F(ExprEvalTest, ConstReference) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  ExpressionEvaluator evaluator;
+  auto tv0 = makeContigTensor(1);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  auto t0 = at::randn({3}, options);
+
+  evaluator.bind(tv0, t0);
+  checkConstEvaluate(evaluator, tv0, t0);
```

Comment:
In this case `tv0` is directly bound so this is not exercising the recursion yet. Try adding some expressions like `auto tv1 = neg(tv0)` then evaluate the unbound value `tv1`.